python/variable.cpp

```diff
@@ -208,22 +226,6 @@ void bind_init_list(py::class_<Variable> &c) {
   c.def(py::init([](const std::array<Dim, 1> &label, const py::list &values,
                     const std::optional<py::list> &variances,
                     const units::Unit &unit, py::object &dtype) {
-          if (scipp_dtype(dtype) == core::dtype<Eigen::Vector3d>) {
```

Comment:
Why is this not required anymore? Moved to `bind_init_list_eigen` as far as I can see, but isn't the different order causing issues somewhere? I remember we had ordering issues in the past. Do the tests you added above cover all cases, or did we overlook anything?

Reply:
Have you tried constructing with values taken from an existing variable with the same dtype?

```python
var = sc.Variable(value=[1, 2, 3], dtype=sc.dtype.vector_3_float64)
 var2 = sc.Variable(value=var.value, dtype=sc.dtype.vector_3_float64) 
```
?
Also without the `dtype` arg for `var2`?

Reply:
no, because `bind_init_list_eigen` is for 0D Variables. I'll rename it to `bind_init_0D_list_eigen` to be more clear.

it is not needed because the list just gets cast to a `py::array` and then forwarded to `doMakeVariable`. I'm not sure it was ever needed in the first place? All python tests pass and docs buils locally...

Reply:
> Have you tried constructing with values taken from an existing variable with the same dtype?
> 
> ```python
> var = sc.Variable(value=[1, 2, 3], dtype=sc.dtype.vector_3_float64)
> var2 = sc.Variable(value=var.value, dtype=sc.dtype.vector_3_float64) 
> ```
> 
> ?
> Also without the `dtype` arg for `var2`?

yes that works

Reply:
It was added in https://github.com/scipp/scipp/pull/631, unfortunately there is no related issue?

Reply:
I assume you have tried this with the Mantid converter, since you added the Instrument View for that purpose? So that can't be it either.

Reply:
> > Have you tried constructing with values taken from an existing variable with the same dtype?
> > ```python
> > var = sc.Variable(value=[1, 2, 3], dtype=sc.dtype.vector_3_float64)
> > var2 = sc.Variable(value=var.value, dtype=sc.dtype.vector_3_float64) 
> > ```
> > 
> > 
> > ?
> > Also without the `dtype` arg for `var2`?
> 
> yes that works

Then one last thing: Does it also work in the 1D case:

```python
var = sc.Variable(dims=['x'], values=bla, dtype=sc.dtype.vector_3_float64)
var2 = sc.Variable(dims=['x'], value=var.values, dtype=sc.dtype.vector_3_float64) 
```
?

Reply:
This doesn't work
```Py
var = sc.Variable(dims=['x'], values=[[1, 2, 3], [4, 5, 6]], dtype=sc.dtype.vector_3_float64)
var2 = sc.Variable(dims=['x'], values=var.values, dtype=sc.dtype.vector_3_float64) 
```
because if you add the `np.array()` it works
```Py
var = sc.Variable(dims=['x'], values=[[1, 2, 3], [4, 5, 6]], dtype=sc.dtype.vector_3_float64)
var2 = sc.Variable(dims=['x'], values=np.array(var.values), dtype=sc.dtype.vector_3_float64) 
```

Reply:
Hmm, doesn't work in `master` either. Bug? In any case, this confirms that nothing was broken by this change?

Reply:
i think it comes from what `.values` returns on a variable of `vector_3_float`. It doesn't return a `np.array`. Maybe it would be nice if we could make it do so?

Reply:
See the longstanding issue #290 --- it may be simpler for `Vector3d`, but in general it is not solvable easily, I think.